quantum/pointing_device/pointing_device_drivers.c

```diff
@@ -172,15 +170,34 @@ report_mouse_t cirque_pinnacle_get_report(report_mouse_t mouse_report) {
                 }
             }
         }
-#    endif
+#        endif
     }
 
-#    ifdef POINTING_DEVICE_GESTURES_CURSOR_GLIDE_ENABLE
+#        ifdef POINTING_DEVICE_GESTURES_CURSOR_GLIDE_ENABLE
 mouse_report_update:
-#    endif
+#        endif
     mouse_report.x = report_x;
     mouse_report.y = report_y;
+#    else
+    // Scale coordinates to arbitrary X, Y resolution
+    cirque_pinnacle_scale_data(&touchData, cirque_pinnacle_get_scale(), cirque_pinnacle_get_scale());
 
+    if (touchData.valid) {
+        if (touchData.buttons & 0x01) {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, true, POINTING_DEVICE_BUTTON1);
+        } else {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, false, POINTING_DEVICE_BUTTON1);
+        }
+        if (touchData.buttons & 0x02) {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, true, POINTING_DEVICE_BUTTON2);
+        } else {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, false, POINTING_DEVICE_BUTTON2);
+        }
+        mouse_report.x = touchData.xDelta;
+        mouse_report.y = touchData.yDelta;
```

Comment:
`uint8_t` to `mouse_xy_report_t` cast, doesn't work with `MOUSE_EXTENDED_REPORT`.
Either explicitly cast to `int8_t` then integer promote, or change the definition in `pinnacle_data_t`.

Reply:
The type should be `int16_t` in `pinnacle_data_t`, nice catch, missed that when copying from my test code.